### PR TITLE
[heritage.gov.hk] New ruleset

### PR DESCRIPTION
**Pull request by Marco129**
New ruleset for [Conserve and Revitalise Hong Kong Heritage](http://www.heritage.gov.hk)

### CODE REVIEW

File: src/chrome/content/rules/heritage.gov.hk.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<ruleset name="heritage.gov.hk">
+	<target host="heritage.gov.hk" />
+	<target host="www.heritage.gov.hk" />
+
+	<rule from="^http://(?:www\.)?heritage\.gov\.hk/"
+		to="https://www.heritage.gov.hk/" />
+
+	<rule from="^http:" to="https:" />
```
Comment (gloomy-ghost) [inline src/chrome/content/rules/heritage.gov.hk.xml]: This `rule` doesn't match any targets, please remove it.

File: src/chrome/content/rules/heritage.gov.hk.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+	<test url="http://heritage.gov.hk/" />
+	<test url="http://www.heritage.gov.hk/" />
```
Comment (gloomy-ghost) [inline src/chrome/content/rules/heritage.gov.hk.xml]: There are implicit tests for targets. Please remove these two.

File: src/chrome/content/rules/heritage.gov.hk.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<ruleset name="heritage.gov.hk">
+	<target host="heritage.gov.hk" />
+	<target host="www.heritage.gov.hk" />
+
+	<rule from="^http://(?:www\.)?heritage\.gov\.hk/"
```
Comment (gloomy-ghost) [inline src/chrome/content/rules/heritage.gov.hk.xml]: `(?:www\.)` and `(www\.)` don't have difference here. I suggest to remove the `?:`

File: src/chrome/content/rules/heritage.gov.hk.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<ruleset name="heritage.gov.hk">
```
Comment (gloomy-ghost) [inline src/chrome/content/rules/heritage.gov.hk.xml]: Note as `Invalid certificate` in a [top comment](https://github.com/EFForg/https-everywhere/search?utf8=%E2%9C%93&q=problematic):

heritage.gov.hk

File: src/chrome/content/rules/heritage.gov.hk.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+	<target host="heritage.gov.hk" />
+	<target host="www.heritage.gov.hk" />
+
+	<rule from="^http://(?:www\.)?heritage\.gov\.hk/"
+		to="https://www.heritage.gov.hk/" />
+
```
Comment (gloomy-ghost) [inline src/chrome/content/rules/heritage.gov.hk.xml]: Please add `<securecookie host=".+" name=".+" />`

**Comment by Marco129** [comment]
PR updated. Thanks!

File: src/chrome/content/rules/heritage.gov.hk.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<!--
+	Problematic subdomains:
+		- (?:www\.)	(cert mismatch: www.heritage.gov.hk)
```
Comment (gloomy-ghost) [inline src/chrome/content/rules/heritage.gov.hk.xml]: I would suggest you to list them like:
```
	Mismatch:
		- www.heritage.gov.hk
```
(Change the spaces to tabs, GitHub automatically transforms them in comments)

**Comment by gloomy-ghost** [comment]
Thanks, merged.

I usually prefer separated commits. This small ruleset is okay, but next time please leave the previous commits when a reviewer has begun reviewing. We can squash them in the end of reviewing.

**Comment by Marco129** [comment]
Sure, no problem. I'll do it in the next time. Thanks!

**Pull request merged**